frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx

```diff
@@ -0,0 +1,387 @@
+    return { name: k, value: v, isRegex: true };
+  });
+  return _.concat([], matches, regexMatches);
+};
+
+/**
+ * Returns new Route object
+ * Ex:
+ * {
+ *   receiver: myNewReceiver,
+ *   match: {
+ *     "severity": "warning",
+ *     "cluster": "myCluster"
+ *   }
+ *   match_re {
+ *     "service": "^(foo1|foo2|baz)$",
+ *   }
+ * }
+ */
+const createRoute = (receiver: AlertManagerReceiver, routeLabels) => {
+  const newRoute = { receiver: receiver.name };
+  _.each(routeLabels, (label) => {
+    _.set(newRoute, [label.isRegex ? 'match_re' : 'match', label.name], label.value);
+  });
+  return newRoute;
+};
+
+/**
+ * Returns new Receiver object
+ * Ex:
+ * {
+ *   name: MyNewReceiver
+ *   pagerduty_configs: {
+ *     routing_key: <integration_key>
+ *   }
+ * }
+ */
+const createReceiver = (formValues, createReceiverConfig, receiverToEdit) => {
+  const receiverConfig = createReceiverConfig(
+    formValues,
+    receiverToEdit && receiverToEdit[formValues.receiverType]
+      ? receiverToEdit[formValues.receiverType][0] // pass in receiver config if editing existing receiver
+      : {},
+  );
+  return {
+    name: formValues.receiverName,
+    [formValues.receiverType]: [{ ...receiverConfig }],
+  };
+};
+
+const subFormFactory = (receiverType) => {
+  switch (receiverType) {
+    case 'pagerduty_configs':
+      return PagerDutyForm;
+    case 'webhook_configs':
+      return WebhookForm;
+    default:
+      return WebhookForm;
+  }
+};
+
+const useFormStore = (initialState) => {
+  const [formValues, setFormValues] = React.useState(initialState);
+
+  const handleChange = (event) => {
+    setFormValues({
+      ...formValues,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  return {
+    formValues,
+    setFormValues,
+    handleChange,
+  };
+};
+
+const getRouteLabelsForEditor = (isDefaultReceiver, receiverToEdit, allRoutes) => {
+  let routeLabels = [];
+
+  if (receiverToEdit) {
+    const receiverRouteLabels = getReceiverRouteLabels(receiverToEdit, allRoutes);
+    routeLabels = !_.isEmpty(receiverRouteLabels)
+      ? routeLabels.concat(receiverRouteLabels)
+      : routeLabels;
+  }
+
+  return !isDefaultReceiver && _.isEmpty(routeLabels)
+    ? [{ name: '', value: '', isRegex: false }]
+    : routeLabels;
+};
+
+const ReceiverBaseForm: React.FC<ReceiverBaseFormProps> = ({
+  obj,
+  titleVerb,
+  saveButtonText,
+  editReceiverNamed,
+}) => {
+  const secret: K8sResourceKind = obj; // Secret "alertmanager-main" which contains alertmanager.yaml config
+  const [errorMsg, setErrorMsg] = React.useState();
+  const [inProgress, setInProgress] = React.useState(false);
+  let config: AlertManagerConfig;
+  if (!errorMsg) {
+    config = getAlertManagerConfig(secret, setErrorMsg);
+  }
+
+  const INITIAL_STATE = {
+    receiverName: '',
+    receiverType: '',
+    routeLabels: [],
+  };
+  _.assign(INITIAL_STATE, PagerDutyForm.getInitialValues(null), WebhookForm.getInitialValues(null));
```

Comment:
Use ES6 features instead of `_.assign`

```ts
  const INITIAL_STATE = {
    receiverName: '',
    receiverType: '',
    routeLabels: [],
    ...PagerDutyForm.getInitialValues(null),
    ...WebhookForm.getInitialValues(null),
  };
```